content/blog/getting-started-with-k8s-part4/index.md

```diff
@@ -0,0 +1,95 @@
+
+You would use a [StatefulSet for](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#using-statefulsets):
+stable and unique network identifiers
+stable and persistent storage
+applications that require ordered deployment and scaling
+applications that require ordered and rolling updates
+
+When a StatefulSet is created, each pod is assigned a unique and persistent identifier. Pods are named numerically starting from 0, e.g., pod-0, pod-1, pod-n. StatefulSets use a [Headless Service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) resource, which sets the `clusterIP` property ton `none`. Pods in a StatefulSet will have an IP address mapped directly to instead of a virtual IP address. This lets other Pods in the cluster directly communicate to an application, as in database connections.
+
+StatefulSets can control the order of starting and stopping Pods. Some applications require a resource to be available before the Pod can be marked healthy. A Pod in a StatefulSet can be mounted to a dedicated [persistent volume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) by declaring a `PersistentVolumeClaim`. If a Pod in a StatefulSet shuts down, the StatefulSet Controller will relaunch a replacement Pod with the same network identifier and reattach it to the persistent Volume.
+
+StatefulSets have two update strategies. The [OnDelete](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#on-delete) strategy does not automatically update the Pods, and they must be manually deleted for the controller to create new Pods. [Rolling updates](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#rolling-updates) automatically deletes and creates a replacement for Pods in a StatefulSet. It creates replacement in the same order as they are deleted, which is from the highest to the smallest numbered Pod. A replacement Pod must be ‘Running and Ready’ before it replaces the preceding Pod.
+
+## Example
+
+Let’s see how this works in practice with a [MariaDB deployment in Kubernetes](https://github.com/pulumi/kubernetes-guides/tree/master/apps/statefulset). The example begins with defining the cluster configuration and the MariaDB configuration using a [configmap](https://kubernetes.io/docs/concepts/configuration/configmap/). We’ll skip over this and go straight to the StatelessSet.
+
+### Headless Service
+
+In the previous section, we described the Headless Service. The key parts are setting the `ClusterIP` to `none`, which creates an IP address and setting the port so that applications can establish a database connection.
```

Comment:
```suggestion
In the previous section, we described the Headless Service. The key parts are setting the `ClusterIP` to `"None"`, which allocates an IP address and port so that applications can establish a database connection.
```